### ISSUE TITLE
Support custom domains for HTTP endpoint URLs

### ISSUE THREAD

**Issue by dylburger** [PipedreamHQ/pipedream#198]
Instead of making HTTP requests to `https://myendpoint.m.pipedream.net`, I'd like to couple a custom domain to a workflow and make requests to `https://my.domain.com`, instead.

**Comment by spiritbro1** [comment]
also if you can add a route to the domain like https://domain.com/webhook so we can check particular route

**Comment by mequelussi** [comment]
This feature would be very important indeed.

**Comment by morozover** [comment]
I think it's enough to add ability to change such type of URL:
https://2c9498e2e746a4051b255f9aca4899e8.m.pipedream.net

To this:
https://mycustomname.m.pipedream.net

**Comment by morozover** [comment]
I think it's enough to add ability to change such type of URL:
https://2c9498e2e778t5987b255f9aca4899e8.m.pipedream.net

To this:
https://mycustomname.m.pipedream.net

**Comment by mequelussi** [comment]
Autocode implemented this feature.

https://autocode.com/community/announcements/autocode-hostnames-api-yourhost-com/

**Comment by tlpriest** [comment]
I need something like https://pipedream.mydomain.tld/ -- not the "myname.pipedream.net" that was suggested. If I want JavaScript to run in a page and make an AJAX call to a pipe, an ad blocker may prevent the latter style from working. I'd pay for this feature such that new workflows with HTTP triggers are https://cname.mydomain.tld/some-string-for-the-endpoint/

**Comment by QB3L** [comment]
Would love the ability to use a custom domain as well.

**Comment by rawkode** [comment]
Came to requets this, glad it's on the list 😃

**Comment by eltonmorais** [comment]
Some Workflows could work as a service. For example, Facebook Pixel API integration. And a few others.

I just want to create my panel, users insert their settings. I generate a url with the proper parameters. And they insert that URL on whereever they want to.

**Comment by dylburger** [comment]
Not yet, but this is still on our backlog! I shared this with the team

**Comment by kolobus** [comment]
Also waiting for this feature. It's needed as a safety measure.

**Comment by Swieter** [comment]
I too have a desire to use a fully custom domain.

**Comment by reedwi** [comment]
Any updates on this capability?

**Comment by Swieter** [comment]
Came back to check on this, I've got another use case I wanted to try this on.

**Comment by Swieter** [comment]
Now that I'm really getting to use Pipedream, I'd really love this ability.  It would be nice to have fully custom domains on my webhooks so I could build or mockup some API endpoints.

**Comment by dylburger** [comment]
@Swieter @reedwi @kolobus Could you tell me more about what controls you'd want? e.g. would you just expect to create your own hostname in Pipedream associated with HTTP endpoints, and then setup DNS on your end? Or are you looking for more complex controls?

**Comment by Swieter** [comment]
I'd want to be able to specify a domain and/or subdomain that I own already.  example.com or api.example.com.
I don't mind having to set up DNS configs to point to pipedream.

Then as I set up webhook or maybe some new trigger, call it API End Point, I'd select my configured domain and then the routes.  So maybe a drop down to pick api.example.com (which was configured in some settings) and then for the specific trigger, I could add "/customers".   Now that i think of it, maybe there are some checkboxes for what verbs are supported - get, post, put, delete

I'd use a code block or some other preconfigured action for processing each verb and composing a response.  

I suppose if you want to get fancy, there may be some action block or config in the trigger for header or api key checking and not proceeding if it doesn't pass muster.

**Comment by dylburger** [comment]
yeah I like that a lot, I think what you're describing can also be extended to any HTTP trigger (absent custom domains). We should support a more clear interface for routing requests for specific verbs / resources to specific workflows / steps.

Sharing this feedback with the team!

**Comment by acron0** [comment]
+1

**Comment by themerk** [comment]
Just wanted to share my use case for this feature. I'm working with a CMS that pushes data to a Pipedream endpoint (effectively using Pipedream to handle a webhook). This gets set up individually for each client using the CMS so I'm having to hardcode the endpoint URL lots of times. Two things that a CNAME or custom domain would help with:

- Look more professional. **webhooks.mycompany.com** instead of the actual Pipedream endpoint URL
- Act as an alias I can point to any Pipedream endpoint and change as needed. I'm paranoid about accidentally deleting or messing with the workflow. I can create a new one quickly, but since the URL is different I'd have to change it in every client system.

**Comment by kolobus** [comment]
@dylburger 
I use pipedream with super "hardcoded" IoT products. And I really want to be able to "hardcode" own domain which is just CNAME to pipedream.
Reason for that is ability to change backend in case something happens - without evacuating IoT devices and reflashing.

**Comment by Paktas** [comment]
3.5 years in the making for one of top requested features... any forecast when this can eventually be born?

**Comment by chris-chameleonbrands** [comment]
+1

**Comment by chris-chameleonbrands** [comment]
It would be also be great if there was support to get of webhooks on the domain. 

IE: 
GET https://my.domain.com/webhooks/list

```
"items": [
    {
    name:"workflow_name",
    url:"https://my.domain.com/webhooks/xyz",
    purl:"https://aNumber.m.pipedream.net"
    }]
```

**Comment by tomngo** [comment]
+1 for this!

**Comment by danielsinewe** [comment]
+1

**Comment by tomngo** [comment]
+1!

**Comment by quest** [comment]
+1

**Comment by dylburger** [comment]
Custom domains are now supported on [the Business plan](https://pipedream.com/pricing) — [see the docs here](https://pipedream.com/docs/workflows/domains/)!

You can configure any domain you want: instead of `https://endpoint.m.pipedream.net`, the endpoint would be available on `https://endpoint.example.com`.

If y'all have any questions about custom domains or our Business + Enterprise plans, feel free to respond here or [book time with us here](https://calendly.com/d/369-489-j8y/chat-with-pipedream?month=2023-08).

**Comment by Swieter** [comment]
Awesome.  Thank you!

Though I can't justify the business plan, I would love to add this as an addon to my existing plan if that was an option.

**Comment by qb3l-pplsi** [comment]
Oh man! Same here, we don't use it enough to go all the way to business yet. Thanks for letting us know @dylburger
